### PR TITLE
Add definitions validation with GA

### PR DESCRIPTION
**Pull request by mpgxvii**
- Add questionnaire validation script (run by Github Actions)
- This validation has 2 parts: 

_1. Questionnaire Schema Validation_
- Added questionnaire definition schema with required properties and so definitions can be validated against those
- This also includes input type validation so only supported input types can be added
- This also includes text input type validation so only supported text input types (text, date, time, etc) can be added

_2. Branching Logic Validation_
- This also checks whether the branching logic (key in definitions: `branching_logic`) is valid
- This has 2 parts: first, it checks whether the logic identifiers/keys are actual field names in the questionnaire
- Second, this checks whether or not the values in the logic are correctly wrapped in quotes.

### CODE REVIEW

**Comment by afolarin** [comment]
I was also thinking we had this GSoC project this also might be an opportunity to build this validation process into any future questionnaire-building interface.
https://github.com/RADAR-base/GSoC/tree/main/project-ideas_2023 project "1A. aRMT Questionnaire Generator"

Though in lieu of that running this as part of the GH Actions is good

File: utilities/validate.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    const breakpoint = /\=|\<|\>|\<=|\>=|\<>/;
+    // Remove equality or inequality symbols to get value
+    const value = rightOperand.split(breakpoint)[1];
+    // Get actual value enclosed in parentheses
+    const valueRegex = /"[^"]*"|'[^"]*'/g;
+    const finalValue = value.match(valueRegex);
+    if (!finalValue) {
+      // Value will be null if not in parenthesis
+      if (!TYPES_REQ_NOT_QUOTED.includes(type))
+        console.warn('WARN: Logic value must be in quotes for logic: ' + logic);
```
[the remark this answers is not in the document]
Comment (mpgxvii) [reply utilities/validate.js:67]: Yes good point, made the changes.

**Review by yatharthranjan — APPROVED** [review]
LGMT, nice bit of fixes already with the action failing 😌

**Comment by afolarin** [comment]
> LGMT, nice bit of fixes already with the action failing relieved

Proof the system works! :+1:

**Pull request merged**